packages/react-reconciler/src/ReactStrictModeWarnings.js

```diff
@@ -281,6 +287,46 @@ if (__DEV__) {
       });
     }
   };
+
+  ReactStrictModeWarnings.recordLegacyContextWarning = (
+    fiber: Fiber,
+    instance: any,
+  ) => {
+    // Dedup strategy: Warn once per component.
+    if (didWarnAboutLegacyContext.has(fiber.type)) {
+      return;
+    }
+
+    if (
+      typeof instance.getChildContext === 'function' ||
+      (fiber.tag === ClassComponent && fiber.type.contextTypes != null) ||
```

Comment:
You shouldn't have the check the `tag` since this is already part of a class-only code path.

Reply:
So should `fiber.tag === ClassComponent` be removed in this case?

Reply:
Yeah